packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/questions.ts

```diff
@@ -113,17 +113,18 @@ export async function getSystemSelectionQuestions(
     );
 
     // Existing system (BackendSystem or Destination) selected,
-    // In future, make the service prompt optional by wrapping in condition `[promptOptions?.serviceSelection?.hide]`
-    questions.push(
-        ...withCondition(
-            getSystemServiceQuestion(
-                connectValidator,
-                systemSelectionPromptNamespace,
-                promptOptions?.serviceSelection
-            ) as Question[],
-            (answers: Answers) => (answers as SystemSelectionAnswers).systemSelection?.type !== 'newSystemChoice'
-        )
-    );
+    if (typeof promptOptions?.serviceSelection?.hide === 'undefined' || !promptOptions?.serviceSelection?.hide) {
```

Comment:
I dont think this additional condition is necessary as:
```
!undefined === true => true
!prompt.undefined === true => true
!prompt.serviceSelection.undefined === true => true
```

Reply:
I'm not even sure how I missed that when I was writing it :). Thanks!

Fixed in: 7635730cc925e9187cf50122427120a32dddda80